client/wsclient.go

```diff
@@ -97,22 +161,37 @@ func (c *wsClient) AgentDescription() *protobufs.AgentDescription {
 }
 
 func (c *wsClient) SetAgentDescription(descr *protobufs.AgentDescription) error {
+	if c.isStopped.Load() {
+		return errStopping
+	}
 	return c.common.SetAgentDescription(descr)
 }
 
 func (c *wsClient) SetHealth(health *protobufs.AgentHealth) error {
+	if c.isStopped.Load() {
+		return errStopping
+	}
 	return c.common.SetHealth(health)
 }
 
 func (c *wsClient) UpdateEffectiveConfig(ctx context.Context) error {
+	if c.isStopped.Load() {
+		return errStopping
+	}
 	return c.common.UpdateEffectiveConfig(ctx)
 }
 
 func (c *wsClient) SetRemoteConfigStatus(status *protobufs.RemoteConfigStatus) error {
+	if c.isStopped.Load() {
+		return errStopping
```

Comment:
This can be addressed separately, but should the HTTP Client be updated to return an error on `SetX` calls after `Stop` is called?

Reply:
Agreed, we should be consistent. I've created an issue to track this here: https://github.com/open-telemetry/opamp-go/issues/206.